lib/pure/uri.nim

```diff
@@ -8,6 +8,35 @@
 #
 
 ## This module implements URI parsing as specified by RFC 3986.
+##
+## A Uniform Resource Identifier (URI) provides a simple and extensible
+## means for identifying a resource. A URI can be further classified
+## as a locator, a name, or both. The term “Uniform Resource Locator”
+## (URL) refers to the subset of URIs.
+##
+## Basic usage
+## ===========
+##
+## Combine URIs
+## -------------
+## .. code-block::
```

Comment:
@ThomasTJdev couldn't this have been a runnableExamples (since it doesn't do networking, no harm in that) ? that way it'd guarantee to stay in sync